lib/stdlib/src/queue.erl

```diff
@@ -453,6 +454,93 @@ fold(Fun, Acc0, {R, F}) when is_function(Fun, 2), is_list(R), is_list(F) ->
 fold(Fun, Acc0, Q) ->
     erlang:error(badarg, [Fun, Acc0, Q]).
 
+%% Check if any item satisfies the predicate, traverse in queue order.
+%%
+%% O(len(Q)) worst case
+-spec any(Pred, Q :: queue(Item)) -> boolean() when
+      Pred :: fun((Item) -> boolean()).
+any(Pred, {R, F}) when is_function(Pred, 1), is_list(R), is_list(F) ->
+    lists:any(Pred, F) orelse
+    lists:any(Pred, lists:reverse(R));
```

Comment:
`any_r(Pred, R);`

``` Erlang
any_r(Pred, [X|R]) when is_list(R) ->
    any_r(Pred, R) orelse (not not Pred(X));
any_r(_Pred, []) ->
    false.
````
Not tested nor compiled. Does not build any garbage like `lists:reverse(R)`.

Reply:
Same question as for `all_r/2`: why the double negation?